GovUkDesignSystem/ModelBinders/GovUkDecimalBinderBase.cs

```diff
@@ -47,13 +41,22 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             // Return if the value is empty
             if (string.IsNullOrEmpty(value))
             {
+                // Raise an error if this property is mandatory or if property type is non-nullable
+                if (errorMessageIfMissing != null || !bindingContext.ModelMetadata.IsReferenceOrNullableType)
+                {
+                    bindingContext.ModelState.TryAddModelError(modelName, errorMessageIfMissing ?? $"{nameAtStartOfSentence} cannot be null or empty");
```

Comment:
The reason we have these custom binders is so that we can use the GovUK validation wording. This default message doesn't follow that as far as I can see. (The closest I can find is "Enter {nameOfField}" on https://design-system.service.gov.uk/components/text-input/)
It seems that using an optional binder on a non-nullable field is a coding error, so maybe we should be throwing an exception if `errorMessageIfMissing == null && !bindingContext.ModelMetadata.IsReferenceOrNullableType`?